src/EnergyPlus/OutputReports.cc

```diff
@@ -1987,52 +1983,67 @@ DetailsForSurfaces( int const RptType ) // (1=Vertices only, 10=Details only, 11
 					ExtConvCoeffCalc = ConvCoeffCalcs( std::abs( Surface( surf ).ExtConvCoeff ) );
 				}
 				if ( Surface( surf ).ExtBoundCond == ExternalEnvironment ) {
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( unit, Format_705, flags ) << "ExternalEnvironment"; }
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( unit, Format_705, flags ) << ExtConvCoeffCalc; }
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( unit, Format_705, flags ) << IntConvCoeffCalc; }
+					if ( DataGlobals::eio_stream ) *DataGlobals::eio_stream << "ExternalEnvironment" << "," << ExtConvCoeffCalc << "," << IntConvCoeffCalc << ",";
```

Comment:
:+1: to all these changes.  The only question I'd bring up is whether we actually want to protect every single call with an if check.  Or if we can collect the `if` check to a more centralized location and abort early if it is null.  Not necessary to change now, but if we're already in here cleaning out the commented lines, then maybe it's worth discussion.